src/GraphQL/Execution/DefaultExecutionStrategySelector.cs

```diff
@@ -0,0 +1,80 @@
+using GraphQLParser.AST;
+
+namespace GraphQL.Execution
+{
+    /// <inheritdoc cref="IExecutionStrategySelector"/>
+    public class DefaultExecutionStrategySelector : IExecutionStrategySelector
```

Comment:
Well, I thought that you won't add all this stuff to fill _default_ registrations. I think that `DefaultExecutionStrategySelector` should stay as simple as possible - just taking a set of registrations and then foreaching over them. Registrations for different operations may be added by extension methods and base set are added in `GraphQLBuilderBase`. Also `_registrationsarrray` actually **may** contain more registrations then 3 (for query, mutation and subscription). It's normal - just peek the first/last - what have the higher priority.

Reply:
Well, the one line of code within the default constructor is required SOMEWHERE.  I can move it into the `DocumentExecuter` so it does `new DefaultExecutionStrategySelector(Array.Empty<ExecutionStrategyRegistration>)` when not using DI, but I thought a constructor here would be more organized, even if the constructor is `internal`.

Reply:
I'm talking about that default selector should not bother with any registration at all. Registractions are external stuff. Think about _default_ as not "default concrete selection of known strategies" but about "default selection implementation" i.e simple foreaching.

Reply:
I don't understand.

Are you saying that it should not provide default instances for Query and Mutation operation types?  That won't work.  DI does not guarantee the order of registrations in the enumerable and there would be no way to know what was a default registration from what was a custom registration.  Or else the DI configuration would be stupid -- like having to provide a factory that examines the enumerables and then adds on the missing operation types or something.  I don't get it.

Reply:
I really don't know what you're suggesting.  I've pretty much copied your suggested code, and only made changes that were necessary to get it to work at all, and you're talking about the selector 'bothering with registrations'.  I suggested a few constructors and then said ok I'd remove them.  So now I don't know what you mean.  You wrote the code, not me.

Reply:
> Are you saying that it should not provide default instances for Query and Mutation operation types? 

Yes.

>  DI does not guarantee the order of registrations in the enumerable and there would be no way to know what was a default registration from what was a custom registration.

Are you sure?

> Or else the DI configuration would be stupid  -- like having to provide a factory that examines the enumerables and then adds on the missing operation types or something. 

Now I do not understand.

I suggest to add default registrations into `GraphQLBuilderBase`.

Reply:
> Are you sure?

Look for yourself.  And when you're done, look for AutoFac.  And so on for every other DI provider.  I'm certainly not going to.  Designing a code pattern that relies on the order of a list of registered implementations by the DI provider would be a very poor coding practice, certainly subject to failure in the failure even if it worked today.

I'm not writing code that is subject to such subjective behavior.

Reply:
I'll check and think about.

Reply:
In MS DI the order is guaranteed.  I don't know about the others.  Still I'm not changing it.  What if it was registered before GraphQLBuilderBase executed?  Code won't work.  Still would be a bad design.

Reply:
Nothing to think about.  Relying on the order of the list is a bad idea.

Reply:
> Nothing to think about. 

😮 There is always something to think about. Even if my proposal is bad (most likely).

Reply:
I'm fine with currrent implementation here after removing other ctor and some code.

Reply:
When you're happy with the API diff and stuff, I'll add the necessary tests, and you can review again before merging.